sympy/integrals/meijerint.py

```diff
@@ -293,7 +293,6 @@ def _mytype(f, x):
         res = []
         for t in types:
             res += list(t)
-        res.sort()
```

Comment:
Apparently it does need to be sorted. It's just a quick lookup, which is then guarded better with real matching. But if you don't sort, then something like `(Heaviside, log)` won't match `(log, Heaviside)`. I suppose we could also use frozensets.